fluent-fallback/src/types.rs

```diff
@@ -120,6 +120,12 @@ impl PartialEq for ResourceId {
     }
 }
 
+impl std::hash::Hash for ResourceId {
```

Comment:
Thought (no action needed): Ah, I'm trying to reason through this, but it seems like we may have some correctness issues around handling multiple resources that have different optional values. Previously the behavior looks like you could add multiple `ResourceId` values with differing `ResourceType`s, but now it will only have one. I'm not sure what the correct resolution is here, or if anything needs to happen. Maybe I should add some tests around this area.

This seems like a low-probability edge case, but should probably be reasoned through.